exoscale/api/compute.py

```diff
@@ -1048,6 +1064,227 @@ def _from_cs(cls, res):
+
+        return res["state"].lower()
+
+    def scale(self, size):
+        """
+        Scale the Instance Pool up or down.
+
+        Parameters:
+            size (int): the number of Compute instance members the pool must manage
+
+        Returns:
+            None
+        """
+
+        if size <= 0:
+            raise ValueError("size must be > 0")
+
+        try:
+            self.compute.cs.scaleInstancePool(
+                id=self.id, zoneid=self.zone.id, size=size
+            )
+        except CloudStackApiException as e:
+            raise APIException(e.error["errortext"], e.error)
+
+        self.size = size
+
+    def update(
+        self,
+        name=None,
+        description=None,
+        instance_template=None,
+        instance_volume_size=None,
+        instance_user_data=None,
+    ):
+        """
+        Update the Instance Pool properties.
+
+        Parameters:
+            name (str): an Instance Pool name
+            description (str): an Instance Pool description
+            instance_template (InstanceTemplate): an instance template to use for
+                Compute instance members
+            instance_volume_size (int): the Compute instance members storage volume
+                size in GB
+            instance_user_data (str): a cloud-init user data configuration to apply to
+                the Compute instance members
+
+        Returns:
+            None
+        """
+
+        instance_template_id = None
+        if instance_template is not None:
+            instance_template_id = instance_template.id
+
+        instance_user_data_content = None
+        if instance_user_data is not None:
+            instance_user_data_content = b64encode(
+                bytes(instance_user_data, encoding="utf-8")
+            )
+
+        try:
+            self.compute.cs.updateInstancePool(
+                id=self.id,
+                zoneid=self.zone.id,
+                name=name,
+                description=description,
+                templateid=instance_template_id,
+                rootdisksize=instance_volume_size,
+                userdata=instance_user_data_content,
+            )
+        except CloudStackApiException as e:
+            raise APIException(e.error["errortext"], e.error)
+
+        if name:
+            self.name = name
+
+        if description:
+            self.description = description
+
+        if instance_template is not None:
+            self.instance_template = instance_template
+
+        if instance_volume_size is not None:
+            self.instance_volume_size = instance_volume_size
+
+    def delete(self):
+        """
+        Delete the Instance Pool.
+
+        Returns:
+            None
+        """
+
+        try:
+            self.compute.cs.destroyInstancePool(id=self.id, zoneid=self.zone.id)
+        except CloudStackApiException as e:
+            raise APIException(e.error["errortext"], e.error)
```

Comment:
Would it be useful to have a "wait" mode here? Other calls doing things async would hang, but this one returns immediately while the request is only queued.

If waiting were the default mode (consistently with other cs calls), it'd simplify the tests as well.

Reply:
I created https://app.clubhouse.io/exoscale/story/7669/python-exoscale-instance-pool-wait-for-a-pool-to-be-destroyed